### PR TITLE
[network] Adding speedtest

### PR DESCRIPTION
**Pull request by clinique**

### CODE REVIEW

File: bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/handler/SpeedTestHandler.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    private @NonNullByDefault({}) SpeedTestConfiguration configuration;
+    private State bufferedProgress = UnDefType.UNDEF;
+
+    public SpeedTestHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        speedTestSocket = new SpeedTestSocket(1500);
```
Comment (J-N-K) [inline bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/handler/SpeedTestHandler.java]: I assume that speedtests are not executed that often. Wouldn't it be better to open and close the `speedTestSocket` only when it is needed (i.e. before l. 87)? You could also set it to `null` on completion and check if it is `null` before starting a new one (or log an error if the old one is still running).

File: bundles/org.openhab.binding.network/src/main/resources/ESH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -100,6 +100,80 @@
 			</parameter>
 		</config-description>
 	</thing-type>
+
```
Comment (J-N-K) [inline bundles/org.openhab.binding.network/src/main/resources/ESH-INF/thing/thing-types.xml]: It looks like there is an issue with formatting in this file.

**Comment by J-N-K** [comment]
Jenkins and Travis fail because you need to add the new dependencies to the feature.

File: bundles/org.openhab.binding.network/src/main/feature/feature.xml [hunk 3]
```diff
@@ -5,6 +5,9 @@
     <feature name="openhab-binding-network" description="Network Binding" version="${project.version}">
         <feature>openhab-runtime-base</feature>
         <feature>openhab-core-model-script</feature>
+        <bundle dependency="true">mvn:fr.bmartel/jspeedtest/1.32.1</bundle>
```
Comment (J-N-K) [inline bundles/org.openhab.binding.network/src/main/feature/feature.xml]:
```suggestion
        <bundle prerequisite="true">wrap</bundle>
        <bundle dependency="true">wrap:mvn:fr.bmartel/jspeedtest/1.32.1</bundle>
        <bundle dependency="true">wrap:mvn:fr.bmartel/http-endec/1.04</bundle>
```

Comment (J-N-K) [reply bundles/org.openhab.binding.network/src/main/feature/feature.xml]: Also please add proper descriptions to both bundles. (see other wrapped bundles)

File: bundles/org.openhab.binding.network/src/main/feature/feature.xml [hunk 4]
```diff
@@ -5,6 +5,10 @@
     <feature name="openhab-binding-network" description="Network Binding" version="${project.version}">
         <feature>openhab-runtime-base</feature>
         <feature>openhab-core-model-script</feature>
+        <bundle prerequisite="true">wrap</bundle>
```
Comment (J-N-K) [inline bundles/org.openhab.binding.network/src/main/feature/feature.xml]: My fault. It‘s 
```suggestion
        <feature prerequisite="true">wrap</feature>
```

Sorry for that.

File: bundles/org.openhab.binding.network/src/main/feature/feature.xml [hunk 5]
```diff
@@ -5,6 +5,10 @@
     <feature name="openhab-binding-network" description="Network Binding" version="${project.version}">
         <feature>openhab-runtime-base</feature>
         <feature>openhab-core-model-script</feature>
+        <feature prerequisite="true">wrap</feature>
+        <bundle dependency="true">wrap:mvn:fr.bmartel/jspeedtest/1.32.1$Bundle-Name=Speed%20Test%20library%20for%20Java&amp;Bundle-SymbolicName=fr.bmartel.jspeedtest&amp;Bundle-Version=1.32.1</bundle>
+	<bundle dependency="true">wrap:mvn:fr.bmartel/http-endec/1.04$Bundle-Name=http-endec&amp;Bundle-SymbolicName=fr.bmartel.http-endec&amp;Bundle-Version=1.04</bundle>
         <bundle start-level="80">mvn:org.openhab.addons.bundles/org.openhab.binding.network/${project.version}</bundle>
```
Comment (J-N-K) [inline bundles/org.openhab.binding.network/src/main/feature/feature.xml]: Somehow you are still missing a dependency.
`<bundle dependency="true">mvn:commons-net/commons-net/3.6</bundle>`

**Comment by J-N-K** [comment]
@davidgraeff any objections?

**Comment by davidgraeff** [comment]
Perfect. I'll merge now.

What I also wanted to do with this binding in the future:
* Copy the WOL code from the samsungtv PR
* Let things go online/offline instead of a channel "online".

**Pull request merged**